docs/guides/advanced-experimentation/anonymous-explainer.md

```diff
@@ -4,110 +4,90 @@ sidebar_position: 5
+The approach above assumes you have relatively dense assignment data. That is, if an anonymous ID is later associated with a user, there will be a corresponding event in the assignment logs. This can be error prone. If a user is assigned on the initial landing page and then converts on a different page, it's easy to imagine the assignment log not having a record with that association. To solve this, we can simply do a one time join between the raw assignment table and another table with the full set of timestamped associations:
 
 ```sql
-SELECT
-   facts.ts
-   , mapping.anonymous_id
-   , facts.user_id
-FROM fact_table as facts
-LEFT JOIN anon_visitor_to_user_mapping AS mapping
-  ON facts.user_id = mapping.user_id
-    AND facts.ts BETWEEN
-      mapping.ts_start_window AND
-      mapping.ts_end_window
+WITH timeboxed_associations AS (
+  SELECT
+        user_id
+      , anonymous_id
+      , lag(ts) OVER (PARTITION BY user_id ORDER BY ts) as valid_from
+      , lead(ts) OVER (PARTITION BY user_id ORDER BY ts) as valid_to
+  FROM user_anonymous_associations
+)
+
+SELECT 
+      a.anonymous_id
+    , ta.user_id
+    , a.experiment
+    , a.variant
+    , a.timestamp
+   FROM assignments a 
+   LEFT JOIN timeboxed_associations ta
+     ON a.anonymous_id = ta.anonymous_id
+    AND ta.valid_from <= a.timestamp
+    AND ta.valid_to > a.timestamp
+     
 ```
 
-## Eppo Setup
+By adding this to your assignment data model you can be sure to capture every association between anonymous IDs and users.
+
+
+## Statistical Considerations
+
+The approach outlined above treats each anonymous ID as a unique subject in the experiment. The astute reader may be concerned about this violating assumptions around independent and identically distributed (i.i.d.) metrics. Specifically, we expect the relationship between two Anonymous IDs associated with the same User ID to be correlated. It might be tempting to instead aggregate by User ID. This would however present two new issues. First, users may experience both variants. It's straightforward to filter these user's out in the Eppo analysis, but this preferentially removes more active users (a users with three associated Anonymous IDs is more engaged than a user with one) and can lead to biased experiment results.
 
-Within Eppo, first create an ‘anonymous’ entity. Feel free to give it whatever name makes the most sense for the organization. This entity will be used to organize both the Assignment SQL definition and the Fact SQL together.
+Second, it's undeniable that most users who have multiple anonymous IDs will never log in on every device. That is, there will inevitably be some Anonymous IDs that are indeed correlated but their relationship is never observed in event logs.
```

Comment:
This is true, but one might think "why not group the anonymous ids you know belong to the same user, even if we can't account for all of them?" It might be worth acknowledging the projected ID approach, even if it falls under the "Using your own existing attribution" approach

Reply:
added a small further justification. i agree we shouldn't shy away from the projected ID thing, but we do mention the BYO approach right below, so hopefully the reader will know they can keep using their own approach if they disagree with ours